numba/core/decorators.py

```diff
@@ -143,34 +143,27 @@ def bar(x, y):
         raise DeprecationError(_msg_deprecated_signature_arg.format('argtypes'))
     if 'restype' in options:
         raise DeprecationError(_msg_deprecated_signature_arg.format('restype'))
-    if options.get('nopython', False) and forceobj:
+    nopython = options.get('nopython', None)
+    if nopython is True and forceobj:
```

Comment:
Does it matter that `nopython` is exactly the `True` object? or does it just need to be truthy?

Reply:
```python
In [4]: @jit(nopython=1, forceobj=True)
   ...: def foo(): return

In [5]: foo()
```

Reply:
Assertion added to enforce `bool` in c3b8c33.